snapcraft/internal/indicators.py

```diff
@@ -64,6 +71,36 @@ def download_requests_stream(request_stream, destination, message=None):
     progress_bar.finish()
 
 
+class UrllibDownlaoder(object):
+    """This is a facility to download an uri with nice progress bars."""
+
+    def __init__(self, uri, destination, message=None):
+        self.uri = uri
+        self.destination = destination
+        self.message = message
+        self.progress_bar = None
+
+    def download(self):
+        urlretrieve(self.uri, self.destination, self._progress_cb)
+
+        if self.progress_bar:
+            self.progress_bar.finish()
+
+    def _progress_cb(self, block_num, block_size, total_length):
```

Comment:
I know I'm being a pita, but can you rename this to _progress_callback?
I have a hardtime with acronyms because my brain is not yet fast enough in english to understand them. I always have to spent a few minutes parsing it. That's the only thing I hate about go, and that's why I love that python code tends to be verbose :)